connector-runtime/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/SpringInboundConnectorFactory.java

```diff
@@ -0,0 +1,53 @@
+package io.camunda.connector.runtime.inbound.lifecycle;
+
+import io.camunda.connector.api.inbound.InboundConnectorExecutable;
+import io.camunda.connector.impl.ConnectorUtil;
+import io.camunda.connector.impl.inbound.InboundConnectorConfiguration;
+import io.camunda.connector.runtime.util.inbound.DefaultInboundConnectorFactory;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * {@link io.camunda.connector.runtime.util.inbound.InboundConnectorFactory} implementation
+ * that supports discovering Connectors defined as Spring beans.
+ * Connectors that are defined as Spring beans are created by Spring.
+ */
+@Component
+public class SpringInboundConnectorFactory extends DefaultInboundConnectorFactory {
+  private final static Logger LOG = LoggerFactory.getLogger(SpringInboundConnectorFactory.class);
+
+  private final Map<String, InboundConnectorExecutable> springConnectorsByType = new HashMap<>();
+
+  @Autowired
+  public SpringInboundConnectorFactory(
+    @Autowired(required = false) List<InboundConnectorExecutable> connectorsFromSpringContext) {
```

Comment:
I believe, you don't need two `@Autowired`. Single one for field should be sufficient.